agent/taskresource/types_unix.go

```diff
@@ -0,0 +1,28 @@
+// +build linux
+
+// Copyright 2017-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package taskresource
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/resources/cgroup"
+	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
+)
+
+// ResourceFields is the list of fields required for creation of task resources
+// obtained from engine
+type ResourceFields struct {
```

Comment:
This won't be needed for other task resources like volume. Is my understanding correct, that if the container doesn't need cgroup, this can be empty in ecsAgent ([here](https://github.com/aws/amazon-ecs-agent/pull/1357/files#diff-2149975f54c050dd31ac9a37f1ef63bcR109))?

Reply:
yes, this won't be needed for volume, but can be used for other resources in the future.